### PR TITLE
updated Readme

### PR DESCRIPTION
**Pull request by Keerthana-Jayakumaran**
Updated the readme file with known information. Please review the same.

Resolves https://github.com/mirage/irmin/issues/981.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,11 +16,29 @@ Irmin is an OCaml library for building mergeable, branchable distributed data st
 ### Documentation
 Documentation can be found online at [https://mirage.github.io/irmin](https://mirage.github.io/irmin)
 
+#### Prerequisites Note 
+  Please ensure to install the latest Opam version to have the latest packages. (As of now,opam's version can be equal to or higher versions than '2.0.4').
```
Comment (pascutto) [inline README.md:20]: The latest version of opam is not necessary, opam 2.0+ should be enough here.

File: README.md [hunk 2]
```diff
@@ -16,11 +16,29 @@ Irmin is an OCaml library for building mergeable, branchable distributed data st
...
+  
+  Check the Ocaml version greater or equal to '4.09.0' with the help of command
```
Comment (pascutto) [inline README.md:22]:
```suggestion
  Check the OCaml version greater or equal to '4.07.0' with the help of command
```

File: README.md [hunk 3]
```diff
@@ -16,11 +16,29 @@ Irmin is an OCaml library for building mergeable, branchable distributed data st
...
+  Check the Ocaml version greater or equal to '4.09.0' with the help of command 
+        
+    opam list   
+
+  If the OCaml version is not the specified version, then use the below commands.
+
+    opam switch create version (eg. version could be 4.09.0)
+    eval $(opam env)
+  
+  We hope, with this prior information you can install the irmin packages successfully.
```
Comment (pascutto) [inline README.md:31]: I don't think this sentence is necessary :)

File: README.md [hunk 4]
```diff
@@ -16,11 +16,29 @@ Irmin is an OCaml library for building mergeable, branchable distributed data st
 ### Documentation
 Documentation can be found online at [https://mirage.github.io/irmin](https://mirage.github.io/irmin)
 
+#### Prerequisites Note
```
Comment (pascutto) [inline README.md:19]: That heading should probably go under `installation`.

File: README.md [hunk 5]
```diff
@@ -16,11 +16,29 @@ Irmin is an OCaml library for building mergeable, branchable distributed data st
 ### Documentation
 Documentation can be found online at [https://mirage.github.io/irmin](https://mirage.github.io/irmin)
 
+#### Prerequisites Note 
+  Please ensure to install the latest Opam version to have the latest packages. (As of now,opam's version can be equal to or higher versions than '2.0.4').
+  
+  Check the Ocaml version greater or equal to '4.09.0' with the help of command 
+        
+    opam list
```
Comment (pascutto) [inline README.md:24]: Perhaps using the `ocaml` command makes more sense and produces less noise? 
```suggestion
    ocaml --version
```

File: README.md (hunk 2 above)
Comment (Keerthana-Jayakumaran) [reply README.md:22]: Hi, @pascutto , but in my case "4.09.0" only worked for me not "4.07.0"

File: README.md (hunk 3 above)
Comment (Keerthana-Jayakumaran) [reply README.md:31]: ok. This can be avoided. If we can show how  to install  opam of specified vesrion.

Comment (pascutto) [reply README.md:31]: How to install opam belongs to the opam documentation and installation instructions. If you want to make a prerequisite section here, the point is to mention:
- Which minimum version of opam is needed.
- Which minimum version of OCaml is needed.
Then redirect the user to the corresponding documentations. We should not duplicate documentation that is available elsewhere.

File: README.md (hunk 2 above)
Comment (pascutto) [reply README.md:22]: Irmin is definitely compatible with 4.07.0. This is checked in CI which you can see [here](https://ci.ocamllabs.io/github/mirage/irmin/commit/7ee44a966820eb46c674d007ca8b9a21e6104d7b).
If you had an issue installing irmin with OCaml 4.07, I suggest you report this back in the issue you opened before (https://github.com/mirage/irmin/issues/981)

**Comment by Keerthana-Jayakumaran** [comment]
@pascutto  . I Updated. Please check the same.

**Comment by samoht** [comment]
Thanks for your contributions!

**Pull request merged**